File: ios/Classes/Model/FirmwareImage.swift
//
//  FirmwareImage.swift
//  flutter_mcumgr
//
//  Created by Tomáš Koranda on 06/09/2020.
//

import Foundation
import McuManager

struct FirmwareImage: Codable {
    
    let slot: Int
    let version:String
    let hash: [UInt8]
    let hashStr: String
    let flags: Dictionary<String, Bool>
    
    init(slot: Int, version: String, hash: [UInt8], flags: Dictionary<String, Bool>) {
        self.slot = slot
        self.version = version
        self.hash = hash
        self.hashStr = Data(hash).hexString
        self.flags = flags
    }
    
    init(image: McuMgrImageStateResponse.ImageSlot){
        self.init(slot: Int(image.slot),
                  version: image.version,
                  hash: image.hash,
                  flags: ["active":image.active,
                          "bootable":image.bootable,
                          "confirmed":image.confirmed,
                          "pending":image.pending,
                          "permanent":image.permanent,])
    }
}

File: android/src/main/java/com/korantom/flutter_mcumgr/FirmwareImage.java
package com.korantom.flutter_mcumgr;

import java.util.HashMap;

import io.runtime.mcumgr.response.img.McuMgrImageStateResponse;

class FirmwareImage {

    final int slot;
    final String version;
    final byte[] hash;
    final String hashStr;
    final HashMap<String, Boolean> flags;

    FirmwareImage(int slot, String version, byte[] hash, String hashStr, HashMap<String, Boolean> flags) {
        this.slot = slot;
        this.version = version;
        this.hash = hash;
        this.hashStr = hashStr;
        this.flags = flags;
    }

    FirmwareImage(final McuMgrImageStateResponse.ImageSlot image) {
        this(image.slot,
                image.version, image.hash,
                StringUtils.toHex(image.hash),
                new HashMap<String, Boolean>() {{
                    put("active", image.active);
                    put("bootable", image.bootable);
                    put("confirmed", image.confirmed);
                    put("pending", image.pending);
                    put("permanent", image.permanent);

                }});
    }
}